app/models/ordergroup.rb

```diff
@@ -148,6 +148,26 @@ def account_updated
     financial_transactions.last.try(:created_on) || created_on
   end
 
+  def self.sort_by_param(param)
+    param ||= "name"
+
+    sort_param_map = {
+      "name" => "name",
+      "name_reverse" => "name DESC",
+      "members_count" => "count(*)",
+      "members_count_reverse" => "count(*) DESC",
+      "last_user_activity" => "max(users.last_activity)",
+      "last_user_activity_reverse" => "max(users.last_activity) DESC",
+      "last_order" => "max(orders.starts)",
+      "last_order_reverse" => "max(orders.starts) DESC"
+    }
+
+    result = self
+    result = result.left_joins(:users).group("groups.id") if param && param.starts_with?("members_count", "last_user_activity")
```

Comment:
you set `param` already in line 152, so there is not need for the `param &&`

Reply:
Oh, that slipped in while I realized the default parameter didn't work. Thanks for pointing it out.